src/Integration/NewConnectedMode/BindingConfiguration.cs

```diff
@@ -45,5 +45,32 @@ public static BindingConfiguration CreateBoundConfiguration(BoundSonarQubeProjec
         public BoundSonarQubeProject Project { get; }
 
         public SonarLintMode Mode { get; }
+
+        public override bool Equals(object obj)
+        {
+            if (obj == null)
+            {
+                return false;
+            }
+
+            if (obj.GetType() != typeof(BindingConfiguration))
+            {
+                return false;
+            }
+
+            BindingConfiguration other = (BindingConfiguration)obj;
+
+            return other.Mode == this.Mode &&
+                other.Project?.Organization?.Key == this.Project?.Organization?.Key &&
+                other.Project?.ProjectKey == this.Project?.ProjectKey &&
+                other.Project?.ServerUri == this.Project?.ServerUri;
+        }
+
+        public override int GetHashCode()
+        {
+            // The only immutable field is Mode.
+            // We don't really expect this type to be used a dictionary key
```

Comment:
May be we should throw instead then?

Reply:
We don't know exactly where _GetHashCode_ is called from except that it is used by collections, so throwing is risky. We have to override _GetHashCode_ because we have overridden _Equals_ so the class behaves as expected by the framework. It just means we have a very poor hashing function.